crates/meroctl/src/cli/context/update.rs

```diff
@@ -81,88 +85,58 @@ impl UpdateCommand {
                 )
                 .await?;
             }
-            Self {
-                context_id,
-                application_id: None,
-                path: Some(path),
-                metadata: None,
-                watch: false,
-                executor: executor_public_key,
-            } => {
-                install_app_and_update_context(
-                    environment,
-                    &client,
-                    multiaddr,
-                    path,
-                    context_id,
-                    None,
-                    &config.identity,
-                    executor_public_key,
-                )
-                .await?;
-            }
-
             Self {
                 context_id,
                 application_id: None,
                 path: Some(path),
                 metadata,
-                watch: true,
                 executor: executor_public_key,
+                ..
             } => {
-                watch_app_and_update_context(
+                let metadata = metadata.map(String::into_bytes);
+
+                let application_id = install_app(
                     environment,
                     &client,
                     multiaddr,
-                    context_id,
-                    path,
-                    metadata.map(String::into_bytes),
+                    path.clone(),
+                    metadata.clone(),
                     &config.identity,
-                    executor_public_key,
                 )
                 .await?;
+                if self.watch {
+                    watch_app_and_update_context(
```

Comment:
this won't update the context unless the application changes

we should update and then watch, see `context create` for reference